hail/src/main/scala/is/hail/lir/Emit.scala

```diff
@@ -127,10 +84,9 @@ object Emit {
             maxStack = depth + 2
 
           setLineNumber(x.lineNumber)
-
-          mn.instructions.add(
-            new MethodInsnNode(INVOKESPECIAL,
-              x.ctor.owner, x.ctor.name, x.ctor.desc, x.ctor.isInterface))
+          mv.visitMethodInsn(
+            INVOKESPECIAL, x.ctor.owner, x.ctor.name, x.ctor.desc, x.ctor.isInterface)
+          instructionCount += 3
```

Comment:
Is this something that should have been there before, or is this new way of doing things changing the instruction count? Is `INVOKE_SPECIAL` 3 instructions?

Reply:
I just added `instructionCount`. Before, the `asm.tree` interface built an in-memory representation of the class file, and in particular it we could interrogate the length of the instruction list of methods. But now we're writing to the class file as we traverse the lir, so we have to keep our own instruction count.

Most lir nodes generate one bytecode instruction, which is handled below the big `match` below. `lir.NewInstanceX` requires 3 bytecode instructions: a `NEW` to allocate a new object, `DUP` to duplicate it, leaving it on the top of the operand stack twice, then `INVOKESPECIAL` to call the constructor on it (after emitting all the arguments to the constructor), which consumes the top one, leaving the bottom one as the result of the `NewInstanceX`.

Reply:
I forgot to mention, `instructionCount` is only used for logging.